icu4c/source/i18n/rulebasedcollator.cpp

```diff
@@ -665,7 +665,7 @@ RuleBasedCollator::setReorderCodes(const int32_t *reorderCodes, int32_t length,
     if(length == 1 && reorderCodes[0] == UCOL_REORDER_CODE_NONE) {
         length = 0;
     }
-    if(length == settings->reorderCodesLength &&
+    if(reorderCodes && length == settings->reorderCodesLength &&
```

Comment:
I think this would be easier to understand if you checked `length > 0` instead of `reorderCodes != nullptr`, for that is what you mean here, right, that `uprv_memcmp()` only should be called if the two lengths are both equal and also larger than zero?